### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by eclaros23**
Created Unit Test Changes: made changes to various tests, for example, post_detail, post_list, post_new, and post_edit.

### CODE REVIEW

**Review by ashcas23 — COMMENTED** [review]
VERY GOOD

File: blog/tests.py [hunk 1]
```diff
@@ -1,3 +1,183 @@
...
+
+  def test_get_postunpublished(self):
+    """If a post is unpublished then the user should not see it"""
+    author = User.objects.create(username = "user1")
+    unpublished_post = Post.objects.create(title = "Post02", text = "Great", author = author)
+    response = self.client.get(f"/post/{unpublished_post.pk}/edit/")
+    self.assertEqual(response.status_code, 200)
+
+  def test_get_loggedout(self):
+    """If user is logged they won't be able to edit posts"""
```
Comment (dchukhin) [inline blog/tests.py]:
```suggestion
    """If user is logged out they won't be able to edit posts"""
```

File: blog/tests.py [hunk 2]
```diff
@@ -1,3 +1,183 @@
...
+    author = User.objects.create(username = "user1") 
+    published_post = Post.objects.create(title = "Post01", text = "Hey", author = author, published_date = now())
+    response = self.client.post(f"/post/{published_post.pk}/edit/", {"title": "new title", "text": "new text"}) 
+    self.assertRedirects(response, f"/post/{published_post.pk}/")
+    published_post.refresh_from_db()
+    self.assertEqual(published_post.title, "new title")
+    self.assertEqual(published_post.text, "new text")
+
+  def test_post_unpublished(self):
+    """If a post is unpublished then the user should not be able to edit it"""
```
Comment (dchukhin) [inline blog/tests.py]:
```suggestion
    """If a post is unpublished then the user should be able to edit it"""
```

File: blog/tests.py [hunk 3]
```diff
@@ -1,3 +1,183 @@
...
+    author = User.objects.create(username = "user1")
+    unpublished_post = Post.objects.create(title = "Post02", text = "Great", author = author)
+    response = self.client.post(f"/post/{unpublished_post.pk}/edit/", {"title": "new title", "text": "new text"})
+    self.assertRedirects(response, f"/post/{unpublished_post.pk}/")
+    unpublished_post.refresh_from_db()
+    self.assertEqual(unpublished_post.title, "new title")
+    self.assertEqual(unpublished_post.text, "new text")
+
+  def test_post_loggedout(self):
+    """If user is logged they won't be able to edit posts"""
```
Comment (dchukhin) [inline blog/tests.py]:
```suggestion
    """If user is logged out they won't be able to edit posts"""
```

File: blog/tests.py [hunk 4]
```diff
@@ -1,3 +1,183 @@
...
+    self.client.logout()
+    author = User.objects.create(username = "user1") 
+    published_post = Post.objects.create(title = "Post01", text = "Hey", author = author, published_date = now())
+    response = self.client.post(f"/post/{published_post.pk}/edit/") 
+    self.assertEqual(response.status_code, 404) 
+    unpublished_post = Post.objects.create(title = "Post02", text = "Great", author = author)
+    response = self.client.post(f"/post/{unpublished_post.pk}/edit/") 
+    self.assertEqual(response.status_code, 404)
```
Comment (dchukhin) [inline blog/tests.py:142]: After this test, could you also add a test for trying to edit a post that doesn't exist (for example, with a URL of `"/post/1000000/edit/"` or `"/post/!!!!!/edit/"`)?

File: blog/tests.py [hunk 5]
```diff
@@ -1,3 +1,183 @@
...
+    self.assertRedirects(response,f"/post/{new_post.pk}/")
+
+  def test_post_loggedout(self):
+    """If user is not logged in they should not create new posts"""
+    self.client.logout() 
+    response = self.client.post("/post/new/")
+    self.assertEqual(response.status_code, 404)
+    response = self.client.post("/post/new/", {"title": "new title", "text": "new text"})
+    self.assertEqual(response.status_code, 404)
+
```
Comment (dchukhin) [inline blog/tests.py:181]: After this test, could you add a test that checks what happens if someone tries to create a post with the same title as another post that already exists?

File: blog/tests.py [hunk 6]
```diff
@@ -1,3 +1,183 @@
...
+  
+
```
Comment (dchukhin) [inline blog/tests.py]: Could you also delete these empty lines at the end of the file?

**Review by dchukhin — COMMENTED** [review]
Good job with writing the tests! I left a couple of comments and some requests for more tests that you could write.

**Review by dchukhin — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**